File: 21560 - Sai Suraj/week 3/1161_Maximum Level Sum of a Binary Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def maxLevelSum(self, root: TreeNode) -> int:
        q,maxl,maxs,lcount=[root],1,root.val,1
        while q:
            levelsum,newq=0,[]
            for n in q:
                if n.left: newq+=[n.left]
                if n.right: newq+=[n.right]
                levelsum+=n.val
            q=newq
            if levelsum>maxs:
                maxl=lcount
                maxs=levelsum
            lcount+=1
        return maxl

File: 21561 - Suvam Tamang/Week 3/1161_Maximum Level Sum of a Binary Tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int maxLevelSum(TreeNode root) {
        int levelSum;
        int maxSum = root.val;
        int maxlevel = 1;
        int level = 1;
        int size = 0;
        
        Queue<TreeNode> q = new LinkedList<>();
        q.add(root);
        
        while (!q.isEmpty()) {
            levelSum = 0;
            size = q.size();
            for(int i=0;i<size;i++){
                TreeNode tempNode = q.poll();
            
                if (tempNode.left != null) {
                    q.add(tempNode.left);
                }

                if (tempNode.right != null) {
                    q.add(tempNode.right);
                }
                levelSum += tempNode.val;
            }
            
            if (levelSum > maxSum) {
                maxSum = levelSum;
                maxlevel = level;
            }
            level++;
        }
        return maxlevel;
    }
}
